sql/parser/ast.go

```diff
@@ -620,11 +653,28 @@ func (node BytesVal) String() string {
 	return string(encodeSQLBytes(scratch[0:0], []byte(node)))
 }
 
+// IntVal represents an integer.
+type IntVal string
```

Comment:
`type IntVal string` seems very odd, especially when `BoolVal` is a bool. This (and `NumVal`) need some explanatory comments, or pointers to comments on this subject elsewhere. At the very least it seems like it would make sense to store numbers in both their string and parsed form so we never have to parse the same string repeatedly.